### PR TITLE
Enhance Cookbooks Report to include Policy file information

### PR DESCRIPTION
**Pull request by mudash**
This PR enhances Cookbooks Report by including Cookbooks that are available to a node via Policy Files. It includes changes to standard out screen report, Txt and CSV formats.

## Description

Today we don't report the cookbooks available on Chef Server that are managed by Policy files. Including these will provide an accurate view of the server and ease to Ops and other users of Chef Analyze.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: pkg/reporting/client_test.go [hunk 1]
```diff
@@ -58,6 +58,51 @@ func (cm CookbookMock) DownloadTo(name, version, localDir string) error {
 	}
 	return cm.desiredDownloadError
 }
+
+//List() (chef.CBAGetResponse, error)
```
Comment (marcparadise) [inline pkg/reporting/client_test.go:62]: nit: cleanup

File: pkg/reporting/cookbooks.go [hunk 2]
```diff
@@ -104,31 +112,50 @@ func (cr *CookbookRecord) NumCorrectable() int {
 	return i
 }
 
-// Sort interface the sorts cookbook records by name.
-type CookbookRecordsByNameVersion []*CookbookRecord
+// Sort interface for cookbook recrods
+// Order by Policy Group, Policy Name, Cookbook Name, Cookbook Version
+type CookbookRecordsBySortOrder []*CookbookRecord
```
Comment (marcparadise) [inline pkg/reporting/cookbooks.go:117]: :+1:

File: pkg/reporting/client_test.go (hunk 1 above)
Comment (mudash) [reply pkg/reporting/client_test.go]: done

File: pkg/reporting/cookbooks.go (hunk 2 above)
Comment (mudash) [reply pkg/reporting/cookbooks.go:117]: Thanks!

**Pull request merged**